lib/commons/rscommons/prism.py

```diff
@@ -0,0 +1,162 @@
+    geom = feature.GetGeometryRef()
+    layer.SetSpatialFilter(geom)
+    # Average the precip measurements
+    precip = statistics.mean([feat.GetField("Mean_Annual_Precip") for feat in layer])
+
+    # Write to huc polygon precip field
+    feature.SetField("Precip_mm", precip)
+    layerPoly.SetFeature(feature)
+    feature = None
+    layer = None
+    layerPoly = None
+    sourceGPKG = None
+    sourcePoly = None
+
+    return precip
+
+
+def calculate_bankfull_width(nhd_flowlines, precip):
+
+    fieldname_bfbuffer = "BFwidth"
+    p = precip / 10.0  # prism data in mm, need cm
+
+    # Load Flowlines
+    driver = ogr.GetDriverByName("ESRI Shapefile")
+    source = driver.Open(nhd_flowlines, 1)
+    layer = source.GetLayer()
+
+    # Write Field BFbuffer_m
+    bfField = ogr.FieldDefn(fieldname_bfbuffer, ogr.OFTReal)
+    layer.CreateField(bfField)
+
+    for feature in layer:
+        a = feature.GetField("TotDASqKm")
+        a = a if a is not None else 0.0
+        w = 0.177 * (a ** 0.397) * (p ** 0.453)
+        feature.SetField(fieldname_bfbuffer, w)
+        layer.SetFeature(feature)
+        feature = None
+
+    layer = None
+    source = None
+
+    return
+
+
+def buffer_by_field(flowlines, field, epsg, conversion=1):
+
+    driver = ogr.GetDriverByName("ESRI Shapefile")
+    source = driver.Open(flowlines, 0)
+    layer = source.GetLayer()
+    in_spatial_ref = layer.GetSpatialRef()
+
+    out_spatial_ref, transform = get_transform_from_epsg(in_spatial_ref, epsg)
+
+    outpolys = []
+    for feature in layer:
+        geom = feature.GetGeometryRef()
+        bufferDist = feature.GetField(field)
+        geom_buffer = geom.Buffer(bufferDist * conversion)
+        geom_buffer.Transform(transform)
+        outpolys.append(wkt_load(geom_buffer.ExportToWkt()))
+
+    # unary union
+    outpoly = unary_union(outpolys)
+    source = None
+
+    return outpoly
+
+
+def bankfull_width_buffer(flowlines, precip):
+
+    p = precip / 10.0  # prism data in mm, need cm
+
+    driver = ogr.GetDriverByName("ESRI Shapefile")
+    source = driver.Open(flowlines, 1)
+    layer = source.GetLayer()
+
+    outpolys = []
+    # Buffer width for each flow line feature
+    for feature in layer:
+        a = feature.GetField("TotDASqKm")
+        a = a if a is not None else 0.0
+        w = 0.177 * (a ** 0.397) * (p ** 0.453)
+        geom = feature.GetGeometryRef()
+        geom_buffer = geom.Buffer(w)
+        outpolys.append(wkt_load(geom_buffer.ExportToWkt()))
+        feature = None
+
+    # unary union
+    outpoly = unary_union(outpolys)
+    source = None
+
+    return outpoly
+
+
+if __name__ == "__main__":
+
+    precip = mean_area_precip(r"D:\NAR_Data\Data\rs_context\17060304\hydrology\WBDHU8.shp", r"D:\NAR\Projects\VBET\Prism_Test02.gpkg")
+
+    calculate_bankfull_width(r"D:\NAR\Projects\VBET\NHDFlowlineCopy.shp", precip)
+
+    polygon = buffer_by_field(r"D:\NAR\Projects\VBET\NHDFlowlineCopy.shp", "BFbuffer")
+
+    print(polygon.area)
```

Comment:
Cab we load this stuff from a .env file? We should avoid any hardcoded paths

Reply:
I don't actually need this anymore. I will make sure I remove it in the future.